engine/gamesys/src/gamesys/components/comp_model.cpp

```diff
@@ -1399,6 +1399,7 @@ namespace dmGameSystem
                 dmVMath::Matrix4 normal_matrix    = dmRender::GetNormalMatrix(render_context, world_matrix);
                 bool has_custom_vertex_attributes = vx_decl != world->m_VertexDeclaration;
 
+                uint32_t material_index = render_item->m_MaterialIndex;
                 vb_end = WriteWorldSpaceVertexData(world, render_context, c, render_item, &material_infos_vertex, vertex_stride, material_index, world_matrix, normal_matrix, has_custom_vertex_attributes, vb_end);
```

Comment:
You were right, it does seem to fix a situation where multiple models are batched, but they have different vertex attributes. Thanks!

Reply:
You're welcome.